starter-code/contact-app/src/app/contact-list/contact-list.component.ts

```diff
@@ -7,18 +7,50 @@ import contactList from '../contacts';
   styleUrls: ['./contact-list.component.css']
 })
 export class ContactListComponent implements OnInit {
-  contacts: Object[];
+  contacts: Object[]; 
   newContact: Object = {};
+  feedbackEnabled = false;
+  processing = false;
 
-  constructor() { }
+  constructor() {}
 
   ngOnInit() {
     this.contacts = contactList;
   }
 
-  addContact(){
-    console.log("Add contact has been called");
-    // add contact to contacts list
-    // clear inputs
+  addContact(form) {
+    this.feedbackEnabled = true;
+    if (form.valid) {   
+      this.processing = true;   
```

Comment:
You need to set processing to false after the contact is pushed. When you work with a database, remember to do this after receiving your response.